docs/design/kata-guest-image-management-design.md

```diff
@@ -49,13 +49,58 @@ Pull the container image directly from the guest VM using `nydus snapshotter` ba
 #### Architecture
 
 The following diagram provides an overview of the architecture for pulling image in the guest with key components. 
-![guest-image-management-architecture](arch-images/guest-image-management-architecture.png)
+```mermaid
+flowchart LR
+    Kubelet[kubelet]--> |1\. Pull image request & metadata|Containerd
+    E-->|2\. Pull image metadata| Containerd
+    Containerd-->Snapshotter[Nydus Snapshotter]
+    Snapshotter-->|3\. Pack image info| Containerd
+    Containerd-->Runtime[Kata Runtime]
+    Runtime-->Hypervisor
+    Hypervisor-->TEE
+    Runtime-->|4\. Pass image info to VM| Agent
+    E[Container Images Registry]-->|6\. Pull image with image info|CDH1
+    subgraph TEE [Virtual Machine]
+      Images[Container Images]-->|7\. Prepare container rootfs|H[Container]
+
+      subgraph CDH [Confidential Data Hub]
+        CDH1[Image Mgmt]
+      end
+
+      CDH-->Images
+      Agent[Kata Agent]-->|5\. Call image pull RPC|CDH
+    end
+```
 
 #### Sequence diagrams
 
 The following sequence diagram depicted below offers a detailed overview of the messages/calls exchanged to pull an unencrypted unsigned image from an unauthenticated container registry. This involves the kata-runtime, kata-agent, and the guest-components’ image-rs to use the guest pull mechanism.
 
-![guest-image-management-details](arch-images/guest-image-management-details.png)
+```mermaid
+sequenceDiagram
+    par Hosts Side
+        Containerd/Kubelet->>runtime.kata_agent: createContainer(ctx,sandbox,c)
+        runtime.kata_agent-->runtime.fs_share_linux: ShareRootFilesystem(ctx,c)
+        runtime.fs_share_linux->>runtime.kata_agent: ShandleVirtualVolumeStorageObject(c,...,KataVolumeType)
+        runtime.kata_agent-->runtime.kata_agent: handleImageGuestPullBlockVolume(c,virtVolume,vol)
+        runtime.kata_agent-->runtime.fs_share_linux: ret:storage
+        runtime.fs_share_linux->>runtime.kata_agent: ret:sharedFile
```

Comment:
Also not quite clear what's happening here: the `-->` arrows don't have a head - what does that mean for the control flow?

Reply:
Fixed with all head